### PR TITLE
Added retries to getOrderCreateWyvernExchangeAddress()

### PR DESCRIPTION
**Pull request by SMLaursen**
Fixes issue #378

### CODE REVIEW

**Comment by alexanderatallah** [comment]
This should be fully cached and non-limited - can you see if you're still being throttled?

File: src/api.ts [hunk 1]
```diff
@@ -126,15 +126,22 @@ export class OpenSeaAPI {
    * Get which version of Wyvern exchange to use to create orders
    * Simply return null in case API doesn't give us a good response
    */
```
Comment (mrbaguvix) [inline src/api.ts:128]: I believe we should update this doc since we added new logic in `try/catch` block.

File: src/api.ts [hunk 2]
```diff
@@ -126,15 +126,22 @@ export class OpenSeaAPI {
...
-      );
-      return null;
+      if (retries <= 0) {
+        this.logger(
+          "Couldn't retrieve Wyvern exchange address for order creation"
+        );
+        return null;
+      } else {
+        await delay(3000);
+        return this.getOrderCreateWyvernExchangeAddress(retries - 1);
```
Comment (mrbaguvix) [inline src/api.ts:143]: can you add a comment about this logic?

**Review by mrbaguvix — CHANGES_REQUESTED** [review]

File: src/api.ts (hunk 1 above)
Comment (SMLaursen) [reply src/api.ts]: Done

File: src/api.ts (hunk 2 above)
Comment (SMLaursen) [reply src/api.ts:145]: Done

**Comment by SMLaursen** [comment]
> This should be fully cached and non-limited - can you see if you're still being throttled?

I'm still being throttled, see my latest comment in https://github.com/ProjectOpenSea/opensea-js/issues/378

**Review by mrbaguvix — APPROVED** [review]

**Comment by ryanio** [comment]
we no longer support wyvern, thanks

**Pull request closed without merging**